### PR TITLE
Bump to JDK 11

### PR DESCRIPTION
**Pull request by oguzhanunlu**

### CODE REVIEW

File: modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala
Comment (chuwy) [inline modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala:136]: Does it actually work? I tried it a week ago and tests were failng with different errors.

Comment (oguzhanunlu) [reply modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala:136]: Initially I hit this bug https://github.com/scala/bug/issues/10418 and wanted to work-around it by this line. Now I've hit another [error](https://github.com/snowplow/enrich/pull/360/checks#step:9:2101) on `Class org.apache.kafka.common.serialization.StringSerializer could not be found` and trying to solve it.

Comment (oguzhanunlu) [reply modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala:136]: > tests were failng with different errors.

Do you refer to unit tests? Or custom tests you performed manually? I'd like to learn more about those failing tests. @chuwy

Comment (chuwy) [reply modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala:136]: I refer to exactly that error you're getting now. I couldn't fix it.

Comment (oguzhanunlu) [reply modules/kafka/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KafkaSource.scala:136]: I just removed that test and CI looks happy now

**Comment by oguzhanunlu** [comment]
@benjben @chuwy if you agree with the changes, I'd like to publish rc1 and see it in action

File: build.sbt
Comment (chuwy) [inline build.sbt:181]: But we didn't delete the actual module?

Comment (oguzhanunlu) [reply build.sbt:181]: This commit deletes `modules/integration-tests`, did you mean something else? @chuwy

Comment (chuwy) [reply build.sbt:181]: No, that's what I meant - sorry didn't notice somehow.

**Comment by chuwy** [comment]
Please create a ticket, @oguzhanunlu explaining changes and force-push the ticket-less commit.

And don't forget to resolve the conflict.

File: modules/beam/src/test/scala/com.snowplowanalytics.snowplow.enrich.beam/enrichments/YauaaEnrichmentSpec.scala
Comment (benjben) [inline modules/beam/src/test/scala/com.snowplowanalytics.snowplow.enrich.beam/enrichments/YauaaEnrichmentSpec.scala:40]: Not sure what this has to do with Java 11

File: modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/outputs/EnrichedEventSpec.scala
Comment (benjben) [inline modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/outputs/EnrichedEventSpec.scala:119]: All of these were taken care of in https://github.com/snowplow/enrich/commit/86536a539e4b5dd659e1d9d07bd76f69e3bef454

I should have created a PR just for this and merge right away. I didn't because it was preventing me from compiling but actually I could have created a new branch just for this and rebase my branch on top of it waiting for the merge. I will do that next time. Sorry!

**Review by benjben — COMMENTED** [review]
Thanks @oguzhanunlu !

I believe that many OSS users use Kafka as a message queue. Would it require a lot of work to get the integration tests to work? If not we might want to invest the time.

File: modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/outputs/EnrichedEventSpec.scala
Comment (oguzhanunlu) [reply modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/outputs/EnrichedEventSpec.scala:119]: No worries @benjben ! Thanks for the empathy!

**Comment by oguzhanunlu** [comment]
Thanks for the comments @chuwy , I just addressed your feedback. @benjben , it took me several hours and I couldn't make that test compile at jdk 11. Feel free to give it a try, but I doubt it is worth the effort, re-writing from scratch would take shorter, if we'd like to offer tests for kafka module.

**Review by chuwy — APPROVED** [review]
Just gave some feedback over Slack. Otherwise - 👍

**Review by benjben — COMMENTED** [review]
LGTM but let's see with Emily if we can remove Kafka integration tests.

**Review by benjben — APPROVED** [review]
Sorry it was my misunderstanding, we are still testing that enrich can read from and write to Kafka.

**Comment by chuwy** [comment]
Feel free to merge, @oguzhanunlu

**Comment by oguzhanunlu** [comment]
Thanks @chuwy , proceeding

**Pull request merged**
